submit.go

```diff
@@ -118,6 +119,9 @@ type submitResponse struct {
 	ReportURL string `json:"report_url,omitempty"`
 }
 
+// regex to catch and substitute ambiguous issue references with explicit ones to the actual repo they are in
+var ambiguousIssueRegex = regexp.MustCompile(`[\s,.](#\d+)`)
```

Comment:
* won't this remove the preceding character?
* will it match the first word in the report?
* will it match foo/#33? I don't think it will, but it should. It feels like it would be better to specify the characters which *cannot* precede the reference.
* will it match #33foo? I think it will, but it shouldn't.

Reply:
> specify the characters which cannot precede the reference.

golang's regexp lib doesn't have negative lookahead due to its guarantee of O(n) efficiency :(

Reply:
that would be lookbehind.

but that doesn't matter, you can use a capturing group and put the captured text back in?

Reply:
> will it match foo/#33? I don't think it will, but it should. It feels like it would be better to specify the characters which cannot precede the reference.

What would foo/#33 map to? owner/repo/foo/#33 doesn't make much sense imo

Reply:
> It feels like it would be better to specify the characters which cannot precede the reference.

In this situation I disagree, it can only be preceded by whitespace or brackets (or nothing if anchored at the beginning), negating that would be much longer, no?

Reply:
I'd have thought it would make sense to follow what github does.

Reply:
Bases what it does based on whether or not a local issue of that ID exists which seems overkill.
foo/#33
test/#123
I think only accepting a `#NNNNN` token on its own, or maybe in brackets should be accepted. The rest should be handled by github UI itself.

Reply:
@richvdh if you have alternate formats that would like this "autocomplete" applied to them then please specify them here, will add to tests and get regex to accept them.

Reply:
> Bases what it does based on whether or not a local issue of that ID exists which seems overkill.

I wasn't suggesting that, fwiw. I was just suggesting that you add `/` to the list of characters which can precede an ambiguous reference (so `foo/#33` would map to `foo/owner/project#33`). It doesn't really matter though.

Reply:
foo/owner/project#33 won't link though so whats the point of translating it?

Reply:
foo/matrix-org/rageshake#33

Reply:
Yes, but that'd get mapped to foo/owner/project#33 which no longer makes sense/works